lib/chef-dk/policyfile/local_lock_fetcher.rb

```diff
@@ -74,7 +74,8 @@ def apply_locked_source_options(options_from_lock)
 
       # @return [String] of the policyfile lock data
       def lock_data
-        FFI_Yajl::Parser.new.parse(content).tap do |data|
+        fetch_lock_data.tap do |data|
```

Comment:
This creates a context switch to read down below to determine what kind of type "data" is, when its more obvious from the in-line code that it is just a plan old Hash returned from JSON parsing.

The `new` call here is also unnecessary, and I'd argue that `#tap` is creating needless overhead in trying to read this as well, so it should just be:

`data = FFI_Yajl::Parser.parse(content)`

Reply:
The fetch_lock_data method was added so the logic in lock_data didn't have to be duplicated here and in remote_lock_fetcher.rb. It's a bit obtuse in this file but I felt made the overall shape a little cleaner and safer.

At any rate if it's safe we can drop the tap.

This was a direct pull over from: #2102 so it lost a bit of the review context that happened there.

Reply:
The usage of `fetch_lock_data` here and override in `RemoteLockFetcher` seems like the right way - It prevents needing to duplicate much of `fetch_lock` in `RemoteLockFetcher`. It's the same pattern  as followed in `ChefServerLockFetcher` and `GitLockFetcher`.